# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/GtkCompositorWidget.h
# Commit: 7ac989d08efb
# Full Hash: 7ac989d08efb12bad409cc29984c58c56a04a6b1
# Author: stransky <stransky@redhat.com>
# Date: 2021-01-22 17:29:50
# Regressor Bug: 1674473
# File Overlap Count: 1
# Description:
#   Bug 1674473 [Linux] SW-WR Clear background when painting popup windows, r=lsalzman
#   
#   Clear alpha from buffer before painting transparent windows by SW Webrender to avoid rendering artifacts.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D102189
# ==============================================================================

diff -r 3046f61ecff2 -r 7ac989d08efb widget/gtk/GtkCompositorWidget.h
--- a/widget/gtk/GtkCompositorWidget.h	Fri Jan 22 08:10:12 2021 +0000
+++ b/widget/gtk/GtkCompositorWidget.h	Wed Jan 20 21:11:48 2021 +0000
@@ -61,6 +61,9 @@
   EGLNativeWindowType GetEGLNativeWindow();
   int32_t GetDepth();
 
+  void ClearBeforePaint(RefPtr<gfx::DrawTarget> aTarget,
+                        const LayoutDeviceIntRegion& aRegion) override;
+
 #if defined(MOZ_X11)
   Display* XDisplay() const { return mXDisplay; }
   Window XWindow() const { return mXWindow; }